src/main/java/org/jenkinsci/lib/envinject/service/EnvInjectActionRetriever.java

```diff
@@ -16,15 +19,28 @@
     //with EnvInjectAction subclasses. Subclasses cannot be casted from
     //all point of Jenkins (classes are not loaded in some points)
     public Action getEnvInjectAction(AbstractBuild<?, ?> build) {
-
+        List<Action> actions;
         if (build == null) {
             throw new NullPointerException("A build object must be set.");
         }
-
-        List<Action> actions;
-        if (build instanceof MatrixRun) {
-            actions = ((MatrixRun) build).getParentBuild().getActions();
-        } else {
+        try {
+            Class<?> matrixClass = Class.forName("hudson.matrix.MatrixRun");
+            if (matrixClass.isInstance(build)) {
+                Method method = matrixClass.getMethod("getParentBuild", null);
+                Object object = method.invoke(build);
+                if (object instanceof AbstractBuild<?, ?>) {
+                    build = (AbstractBuild<?, ?>) object;
+                }
+            }
+        } catch (ClassNotFoundException e) {
+            LOGGER.log(Level.FINEST, String.format("hudson.matrix.MatrixRun is missed", e));
+        } catch (NoSuchMethodException e) {
+            LOGGER.log(Level.WARNING, String.format("The method getParentBuild does not exist for hudson.matrix.MatrixRun", e));
+        } catch (IllegalAccessException e) {
+            LOGGER.log(Level.WARNING, String.format("There was a problem in the invocation of getParentBuild in hudson.matrix.MatrixRun", e));
+        } catch (InvocationTargetException e) {
+            LOGGER.log(Level.WARNING, String.format("There was a problem in the invocation of getParentBuild in hudson.matrix.MatrixRun", e));
+        } finally {
             actions = build.getActions();
```

Comment:
🐛 I would take this out of the finally. If you have a RuntimeException you won't want to execute this statement.